include/swift/Runtime/Concurrency.h

```diff
@@ -900,6 +918,20 @@ SerialExecutorRef swift_task_getCurrentExecutor(void);
 SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
 SerialExecutorRef swift_task_getMainExecutor(void);
 
+/// Return the generic (default global concurrent) executor reference.
+SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
+SerialExecutorRef swift_task_getGenericExecutor(void);
+
+/// Return the "undefined" task executor reference.
+///
+/// It can be used used to store a record indicating that
+/// there is no task executor preference in the current scope.
+SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
+TaskExecutorRef swift_task_getUndefinedTaskExecutor(void);
+
+SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
+TaskExecutorRef swift_task_getPreferredTaskExecutor(void);
```

Comment:
This really needs documentation explaining the constraints on how it can be used.

Reply:
Added in: https://github.com/apple/swift/pull/68793/commits/10cadfa2a876bbe3d9a3ae736b186dc6425cc08a I think that covers the "be careful" side of things. Anything more we should add?